src/lug/Window/Window.cpp

```diff
@@ -0,0 +1,68 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// src/lug/Window/Window.cpp
+// c'est l� que �a devient int�ressant, on va utiliser notre impl�mentation plateforme sp�cifique
+// d'apr�s ce code, l'impl�mentation plateforme sp�cifique doit d�finir "create" et "processEvents"
+// WARNING: Le code ne respecte pas forc�ment notre guideline, c'est juste un exemple !
+//
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include <lug/Window/Window.hpp>
+#include <iostream>
+
+// on inclus la d�claration de lug::Window::priv::WindowImpl
+#if defined(LUG_SYSTEM_WINDOWS)
+
+#include <lug/Window/Win32/WindowImplWin32.hpp>
+
+#elif defined(LUG_SYSTEM_LINUX)
+
+#include <lug/Window/Unix/WindowImplX11.hpp>
+
+#elif defined(LUG_SYSTEM_ANDROID)
+
+#include <lug/Window/Android/WindowImplAndroid.hpp>
+#include "..\..\..\..\include\lug\Window\Win32\WindowImplWin32.hpp"
+
+#endif
+
+lug::Window::Window::Window() : _impl(new lug::Window::priv::WindowImpl())
+{
+}
+
+std::unique_ptr<lug::Window::Window> lug::Window::Window::create(int width, int height, const std::string & title, uint32_t style)
+{
+	std::unique_ptr<Window> win = std::make_unique<Window>();
```

Comment:
Just use `std::make_unique<Window>(new Window());` if the constructor is protected or private ;)